segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -83,9 +86,11 @@
     private static final int HIGH_WATER_MARK = 128 * 1024;
     private static final int LOW_WATER_MARK = 64 * 1024;
     private static final StatsLogger STATS_LOGGER = MetricsProvider.createStatsLogger("segmentstore");
-    private static final DynamicLogger DYNAMIC_LOGGER = MetricsProvider.getDynamicLogger();
     private static final OpStatsLogger WRITE_STREAM_SEGMENT = STATS_LOGGER.createStats(SEGMENT_WRITE_LATENCY);
     private static final String EMPTY_STACK_TRACE = "";
+    @VisibleForTesting
+    @Setter(AccessLevel.PACKAGE)
+    private DynamicLogger dynamicLogger = MetricsProvider.getDynamicLogger();
```

Comment:
This is not thread safe and makes the class non-immutable.
Either:
1. Make this final, access it via a getter, then in your tests you subclass this class and override the getter.
2. Create a getter and setter for this and either store it in an AtomicReference or make the getter-setter synchronized.

Reply:
Technically, it is not thread safe. But there is an easy fix. AppendProcessor already has a constructor that is specific to tests. We can make this value final and assign it in the constructor, and add an argument to the constructor for tests that provides it.

Reply:
Yes, mutable object is of concern... fixed.